packages/@dcos/sorting/index.ts

```diff
@@ -0,0 +1,61 @@
+export enum ComparisonResult {
+  BIGGER = 1,
+  EQUAL = 0,
+  SMALLER = -1
+}
+export type Comparator<T> = (a: T, b: T) => ComparisonResult;
+export type Lense<T> = (data: T) => any;
+
+export function sort<T>(comps: Array<Comparator<T>>, data: T[]): T[] {
+  return [...data].sort((a, b) => {
+    for (const comperator of comps) {
+      const result = comperator(a, b);
+
+      if (result !== 0) {
+        return result;
+      }
+    }
+
+    return 0;
+  });
+}
+
+export function reverseComparator<T>(comparator: Comparator<T>): Comparator<T> {
+  return (a, b) => {
```

Comment:
Why not just ?
return (a,b) => -1 * comparator(a,b);

Reply:
Had a discussion with @juliangieseke about this and we did not want to spread `-0` values into our system as we don't know if this might have side-effects. I know, there is no `-0` mathematically, but in JS there is.

Reply:
I see. I think `-0` would not be an issue as:
`-0` === '`0`. -0 also makes sense mathematically, it is just rarely written. 

But I  understand not wanting to introduce the confusion so 👍

Reply:
More as some funny things about -0. https://en.wikipedia.org/wiki/Signed_zero